test-framework/src/main/java/org/apache/brooklyn/test/framework/TestHttpCallImpl.java

```diff
@@ -109,6 +130,49 @@ public Integer get() {
         }
     }
 
+    private HttpRequestBase createHttpMethod(HttpMethod method, String url, Map<String, String> headers, String body) throws Exception {
+        switch (method) {
+            case GET:
+                HttpTool.HttpGetBuilder httpGetBuilder = new HttpTool.HttpGetBuilder(new URI(url));
```

Comment:
This is a code-smell that we need to do lots of identical code for each case, rather than being able to use the `HttpTool` in a simpler way (e.g. passing in the method to a generic request builder, perhaps).